src/x/xPadding.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * This function performs a circular shift to a new array
+ * Positive values of shifts will shift to the right and negative values will do to the left
+ * @example xRotate([1,2,3,4],1) -> [4,1,2,3]
+ * @example xRotate([1,2,3,4],-1) -> [2,3,4,1]
+ * @param {Array} array - the array that will be rotated
+ * @param {object} [options={}]
+ * @param {string} [options.algorithm='value'] none, value, circular, duplicate
+ * @param {number} [options.size=0] padding size before and after
+ * @param {number} [options.value=0] value to use for padding (if algorithm='value')
+ * @return {Array}
+ */
+export function xPadding(array, options = {}) {
+  const { size = 0, value = 0, algorithm = 'value' } = options;
+  let result = new Float64Array(array.length + size * 2);
```

Comment:
Maybe we can validate the input with `import xCheck from './xCheck';`
```suggestion
  xCheck(array);
  let result = new Float64Array(array.length + size * 2);
```